core/templates/pages/exploration-player-page/services/content-translation-manager.service.ts

```diff
@@ -247,9 +258,10 @@ export class ContentTranslationManagerService {
           let ruleInputValue = rule.inputs[key];
           if (this._isTranslatableObject(ruleInputValue)) {
             const writtenTranslation = writtenTranslations.translationsMapping[
-              ruleInputValue.contentId][languageCode];
+              <string>ruleInputValue.contentId][languageCode];
             // There must be at least one translation.
-            if (writtenTranslation.translation.length === 0) {
+            if (!writtenTranslation ||
+            writtenTranslation.translation.length === 0) {
```

Comment:
Why is the logic in if check changed ? The condition will pass if `writtenTranslation` is null, undefined or `''`.

Reply:
```suggestion
            if (
              !writtenTranslation ||
              writtenTranslation.translation.length === 0
            ) {
```